tests/cxx/single_inheritance.cpp

```diff
@@ -0,0 +1,98 @@
+/*@
+predicate A_pred(A *a; int i) =
+	a != 0 &*& a->i |-> i;
+@*/
+
+struct A {
+    int i;
+    
+    A()
+    //@ requires true;
+    //@ ensures A_pred(this, _);
+    {
+    	//@ close A_pred(this, _);
+    }
+    
+    ~A()
+    //@ requires A_pred(this, _);
+    //@ ensures true;
+    {
+    	//@ open A_pred(this, _);
+    }
+    
+    void incr()
+    //@ requires A_pred(this, ?i);
+    //@ ensures A_pred(this, i + 1);
+    {
+    	++i;
+    }
+};
+
+/*@
+predicate B_pred(B *b; int a_i, int b_i) =
+	A_pred(b, a_i) &*& b->i |-> b_i;
+@*/
+
+struct B : A {
+    int i;
+
+    B() : i(1) 
```

Comment:
Also test the case where the derived constructor explicitly calls the base constructor with some (nonzero number of) arguments?

Reply:
Also test the case where a derived constructor delegates to some other derived constructor which then calls the base constructor with some arguments (that use the derived constructor's parameters)?

Reply:
Added delegating calls